tensorflow/lite/micro/kernels/xtensa/pooling_int8.cc

```diff
@@ -0,0 +1,338 @@
+
+    int required_scratch = xa_nn_maxpool_getsize(
+        depth, PREC_8, PREC_8, input_height, input_width, params->filter_height,
+        params->filter_width,
+        params->stride_width,                    // x_stride,
+        params->stride_height,                   // y_stride,
+        data->reference_op_data.padding.width,   // x_padding,
+        data->reference_op_data.padding.height,  // y_padding,
+        output_height, output_width, 0 /* NHWC inpput */, 0 /* NHWC output */);
+
+    if (required_scratch <= 0) {
+      MicroPrintf("Maxpool: xa_nn_maxpool_getsize failed");
+      return kTfLiteError;
+    }
+
+    TF_LITE_ENSURE_STATUS(context->RequestScratchBufferInArena(
+        context, required_scratch, &(data->scratch_tensor_index)));
+  }
+
+  micro_context->DeallocateTempTfLiteTensor(input);
+  return kTfLiteOk;
+}
+
+TfLiteStatus MaxEvalQuantizedHifi(TfLiteContext* context, TfLiteNode* node,
+                                  TfLitePoolParams* params,
+                                  const XtensaOpDataPooling* data,
+                                  const TfLiteEvalTensor* input,
+                                  TfLiteEvalTensor* output) {
+  const RuntimeShape& input_shape = tflite::micro::GetTensorShape(input);
+  const RuntimeShape& output_shape = tflite::micro::GetTensorShape(output);
+  const int batches = MatchingDim(input_shape, 0, output_shape, 0);
+  const int depth = MatchingDim(input_shape, 3, output_shape, 3);
+  const int input_height = input_shape.Dims(1);
+  const int input_width = input_shape.Dims(2);
+  const int output_height = output_shape.Dims(1);
+  const int output_width = output_shape.Dims(2);
+
+  void* p_scratch = static_cast<void*>(
+      context->GetScratchBuffer(context, data->scratch_tensor_index));
+
+  const int8_t* inp_data_ptr = tflite::micro::GetTensorData<int8_t>(input);
+  int8_t* out_data_ptr = tflite::micro::GetTensorData<int8_t>(output);
+
+  for (int batch = 0; batch < batches; ++batch) {
+    TF_LITE_ENSURE_EQ(
+        context,
+        xa_nn_maxpool_8(
+            &out_data_ptr[output_height * output_width * depth * batch],
+            const_cast<int8_t*>(
+                &inp_data_ptr[output_height * output_width * depth * batch]),
+            input_height, input_width, depth, params->filter_height,
+            params->filter_width, params->stride_width, params->stride_height,
+            data->reference_op_data.padding.width,
+            data->reference_op_data.padding.height, output_height, output_width,
+            0, 0, p_scratch),
+        0);
+  }
+
+  const int out_length = batches * output_height * output_width * depth;
+  TF_LITE_ENSURE_EQ(
+      context,
+      xa_nn_vec_activation_min_max_8_8(
+          out_data_ptr, out_data_ptr, data->reference_op_data.activation_min,
+          data->reference_op_data.activation_max, out_length),
+      0);
+
+  return kTfLiteOk;
+}
+
+#endif  // defined(HIFI5)
+
+namespace {
```

Comment:
Please keep the anonymous namespace at the top of the file.

Reply:
done